src/avk.cpp

```diff
@@ -2348,28 +2348,38 @@ namespace avk
 			// We have to create a (somewhat temporary) staging buffer and transfer it to the GPU
 			// "somewhat temporary" means that it can not be deleted in this function, but only
 			//						after the transfer operation has completed => handle via sync
-			auto stagingBuffer = root::create_buffer(
-				mPhysicalDevice, mDevice, mBuffer.allocator(),
-				AVK_STAGING_BUFFER_MEMORY_USAGE,
-				vk::BufferUsageFlagBits::eTransferSrc,
-				generic_buffer_meta::create_from_size(dataSize)
-			);
-			stagingBuffer->fill(aDataPtr, 0, sync::wait_idle()); // Recurse into the other if-branch
+			// We need to take care though, to not try to allocate buffer of size zero here.
+			// If dataSize is zero, skip staging buffer creation and the copy command, but still
+			// process the synchronization calls, as user code may rely on those.
+			buffer stagingBuffer;
+			if (dataSize != 0) {
+				stagingBuffer = root::create_buffer(
+					mPhysicalDevice, mDevice, mBuffer.allocator(),
+					AVK_STAGING_BUFFER_MEMORY_USAGE,
+					vk::BufferUsageFlagBits::eTransferSrc,
+					generic_buffer_meta::create_from_size(dataSize)
+				);
+				stagingBuffer->fill(aDataPtr, 0, sync::wait_idle()); // Recurse into the other if-branch
+			}

```

Comment:
Shouldn't we issue a warning in this case (i.e. using `AVK_LOG_WARNING`)?!
Or is this something that could also happen in a non-warning-worthy use case?

Reply:
I'd prefer no warning, if the data size is zero, as this would complicate client code.
E.g. in my current project, I fill a number of buffers each frame, and it is not uncommon that for one of them the size is 0.